### PR TITLE
Video embeds

### PR DESCRIPTION
**Pull request by kyle-beattie**
Using embedly for all videos:

- Existing embedly videos still work (Youtube, Vimeo etc)
- Made explainer, Xero & about us videos work with embedly
- Removed inline embedly script
- Removed vimeo link
- Refactored video JS to async load embedly

[See the story](https://vimaly.com/#j8mqm/t/www_CWV_Figure_out_what_to_do_with_video_embeds./47evbKraDC0XeCMOD)

### CODE REVIEW

File: source/js/src/video.js.es6 [hunk 1]
```diff
@@ -1,52 +1,100 @@
...
-    const vimeo = new Player(vimeoSrc, {
-      color: 'F5A623',
-      title: false,
-      byline: false,
-      portrait: false
-    });
+      require(['//cdn.embedly.com/widgets/platform.js'], function () {
+        embedly('defaults', {
+          cards: {
+            key: '29eeda9e10194668b8297779af3ec8eb',
```
Comment (kylorhall) [inline source/js/src/video.js.es6:32]: Is this key the same for production and testing—does it matter?

File: source/layouts/partials/_head.html.erb [hunk 2]
```diff
@@ -40,9 +40,6 @@
   <link rel="preload" href="<%= asset_url("fonts/opensans-light-webfont-webfont.woff") %>" as="font" type="font/woff2" crossorigin>
   <link rel="preload" href="<%= asset_url("fonts/opensans-regular-webfont-webfont.woff") %>" as="font" type="font/woff2" crossorigin>
 
-  <!-- Preload some Javascript files we will load in the future. -->
```
Comment (kylorhall) [inline source/layouts/partials/_head.html.erb:43]: If you do see Chrome say there's a preload available, add it back in 😉…

I have no idea what Embedly does..

File: source/partials/video/_tony.html.erb [hunk 3]
```diff
@@ -4,6 +4,12 @@
...
-
-  <div data-vimeo-id="116498595" data-vimeo-width="500" data-vimeo-height="281"></div>
+  <iframe
+    class="embedly-embed"
+    frameborder="0"
+    scrolling="no"
+    allowfullscreen
+    src="//cdn.embedly.com/widgets/media.html?src=https%3A%2F%2Fplayer.vimeo.com%2Fvideo%2F116498595&url=https%3A%2F%2Fvimeo.com%2F116498595&type=text%2Fhtml&schema=vimeo"
+    width="1920"
+    height="800"></iframe>
```
Comment (kylorhall) [inline source/partials/video/_tony.html.erb:14]: LGTM 🤞

File: source/js/src/video.js.es6 [hunk 4]
```diff
@@ -1,52 +1,100 @@
...
+   */
+  function initialize () {
+    if (initialized) return;
+
+    const staticVideoElements = document.querySelectorAll('.o-video_wrapper')
+
+    if (staticVideoElements.length) {
+      loadEmbedly(staticVideoElements).then(function (videos) {
+        videos.forEach(function(video) {
+          video.playButton.addEventListener('click', e => {
```
Comment (kylorhall) [inline source/js/src/video.js.es6:71]: This LGTM…I think I understand what's going on here!

File: source/js/src/video.js.es6 [hunk 5]
```diff
@@ -1,52 +1,100 @@
...
+   */
+  const loadEmbedly = function(staticVideoElements) {
+    return new Promise((resolve, reject) => {
+      if (loaded) resolve(videos);
 
-    const vimeoSrc = document.querySelector('[data-vimeo-url], [data-vimeo-id]');
-    if (!vimeoSrc) return false;
-    initialized = true;
+      if (staticVideoElements && staticVideoElements.length) {
+        for (let i = 0; i < staticVideoElements.length; i++) {
```
Comment (kylorhall) [inline source/js/src/video.js.es6:18]: 95% sure `for …` is cross-browser compatible.  But this is a `NodeArray` or whatever it's called…so worth having it tested in IE11.

This seems to be aware of that 😉 given you didn't do `staticVideoElements.forEach`.

File: source/js/src/video.js.es6 [hunk 6]
```diff
@@ -1,52 +1,100 @@
 //= require "../lib/require.2.3.6.min.js"
 
-// NOTE: We use require.js to load Vimeo, then initialize Vimeo.
-// NOTE: …/player.js is preloaded in the <head>!
-require(['https://player.vimeo.com/api/player.js'], function(Player) {
+// Videos
+;(function() {
```
Comment (kylorhall) [inline source/js/src/video.js.es6:4]: I had a brief look and I think this whole function looks great.  Can't say I 100% follow it without really digging in, but yeah.

**Review by kylorhall — APPROVED** [review]
I'll call this a 👍—let me know if you need any help on a specific scenario/area, but I think this is pretty perfect, seems IE11 compatible, async, etc to me..

File: source/js/src/video.js.es6 (hunk 5 above)
Comment (kyle-beattie) [reply source/js/src/video.js.es6:18]: Precisely 👌

**Comment by kyle-beattie** [comment]
@kylorhall I've refactored this so that embedly is only loaded when you click the play button on one of the static videos, e.g the home page one.

Also, I've used the `loading="lazy"` attribute on the iframes for the static videos: https://caniuse.com/loading-lazy-attr. Limited browser support but it's a bit of a perf bump in Chrome at least

File: source/partials/video/_explainer.html.erb [hunk 7]
```diff
@@ -4,6 +4,13 @@
...
     </a>
   </div>
-
-  <div data-vimeo-id="57338416" data-vimeo-width="500" data-vimeo-height="281"></div>
+  <iframe
+    class="embedly-embed"
+    frameborder="0"
+    scrolling="no"
+    allowfullscreen
+    loading="lazy"
```
Comment (kylorhall) [inline source/partials/video/_explainer.html.erb:12]: Yeah, this makes sense.  I don't see a reason not to, if it works.  👍

**Review by kylorhall — APPROVED** [review]
Still 👍 LGTM assuming QA/Marketing are happy, works as best as I can understand from a technical review perspective.

**Comment by kyle-beattie** [comment]
Slightly better results:


**Before:**
staging-before


**After:**
Screen Shot 2021-06-01 at 3 32 39 PM

**Comment by kylorhall** [comment]
@kyle-beattie I think that's great.  Yellow => Green is probably big enough.  Random arbitrary tweet that provides no proof shows it's an impactful change (if this impacts painting): https://twitter.com/JohnMu/status/1395798952570724352?s=19

**Pull request merged**